enumerable_methods.rb

```diff
@@ -0,0 +1,65 @@
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return unless block_given?
+
+    index = 0
+    for i in self
+      yield i, index
+      index += 1
+    end
+  end
+
+  def my_select
+    arr = []
+    my_each { |n| arr.push(n) if yield n } if block_given?
+    arr
+  end
+
+  def my_all?
+    my_select { |n| yield n }.length == length if block_given?
+  end
+
+  def my_any?
+    my_select { |n| yield n }.length.positive? if block_given?
+  end
```

Comment:
- [ ] enumerables #my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
      Failure/Error: expect(true_array.my_any?).to be true_array.any?
- [ ] enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true
      Failure/Error: expect(false_array.my_any?).to be false_array.any?

- [ ] enumerables #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric
      Failure/Error: expect(array.my_any?(Numeric)).to be array.any?(Numeric)

- [ ] enumerables #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer
      Failure/Error: expect(words.my_any?(Integer)).to be words.any?(Integer)
- [ ] enumerables #my_any? when a Regex is passed as an argument returns true if any of the collection matches the Regex
      Failure/Error: expect(words.my_any?(/d/)).to be words.any?(/d/)
- [ ] enumerables #my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
      Failure/Error: expect(words.my_any?(/z/)).to be words.any?(/z/)

- [ ]  enumerables #my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
      Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')

- [ ] enumerables #my_any? when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern
      Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')